src/devices/OneWire/OneWireBus.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Iot.Device.OneWire
+{
+    public class OneWireBus
+    {
+        internal OneWireBus(string deviceId)
+        {
+            DeviceId = deviceId;
+        }
+
+        public string DeviceId { get; }
+
+        public IEnumerable<OneWireDevice> EnumerateDevices(int family = -1)
+        {
+            return OneWireController.EnumerateDevices(this, family);
+        }
+
+        public Task ScanForDevicesAsync(int numDevices = 5, int numScans = -1)
```

Comment:
personally I'm not fan of async APIs for stuff which is fairly fast - would be nice to have a sync and async versions if you strongly feel about having async.

Reply:
This is WIP. I didn't find a specific notification of when the Linux driver is done searching for devices. I will have to read the kernel driver in more detail to find a possible trigger of the search ending. Worst case is that I have to read the driver timeout and poll a bit after that to see if it is done.
Also, there is a continuous search/scan mode in Linux. Not sure if that should have its' own API. Now I just used the driver one using -1 attempts for continuous.
And if using scan with multiple searches, should the operation finish after the first one when the devices are most likely found or the last one?
As for async, I'm a fan ;) And in this case the single search timeout is 10 seconds (read only) in Linux. Which is enough for motivating async. But as I said, I don't wait that long yet.

Reply:
np :) Hardware is weird and some devices are more `special` than others - 1-wire seems to be like that 😄